Source/Core/AbsyExpr.cs

```diff
@@ -3837,6 +3843,206 @@ public T Dispatch<T>(IAppliableVisitor<T> visitor)
     }
   }
 
+  public class FieldAccess : IAppliable
+  {
+    private IToken _tok;
+
+    private string fieldName;
+    
+    private DatatypeSelector selector;
+
+    public IToken tok
+    {
+      get
+      {
+        Contract.Ensures(Contract.Result<IToken>() != null);
+        return this._tok;
+      }
+      set
+      {
+        Contract.Requires(value != null);
+        this._tok = value;
+      }
+    }
+    
+    public string FieldName
+    {
+      get
+      {
+        Contract.Ensures(Contract.Result<IToken>() != null);
+        return this.fieldName;
+      }
+      set
+      {
+        Contract.Requires(value != null);
+        this.fieldName = value;
+      }
+    }
+    
+    public DatatypeSelector Selector
+    {
+      get
+      {
+        Contract.Ensures(Contract.Result<IToken>() != null);
+        return this.selector;
+      }
+      set
+      {
+        Contract.Requires(value != null);
+        this.selector = value;
+      }
+    }
+
+    [ContractInvariantMethod]
+    void ObjectInvariant()
+    {
+      Contract.Invariant(this._tok != null);
+    }
+
+    public FieldAccess(IToken tok, string fieldName)
+    {
+      Contract.Requires(tok != null);
+      this._tok = tok;
+      this.fieldName = fieldName;
+    }
+    
+    public FieldAccess(IToken tok, DatatypeSelector selector)
+    {
+      Contract.Requires(tok != null);
+      this._tok = tok;
+      this.selector = selector;
+      this.fieldName = selector.OriginalName;
+    }
+
+    public string FunctionName
+    {
+      get
+      {
+        Contract.Ensures(Contract.Result<string>() != null);
+
+        return "field-access";
+      }
+    }
+
+    [Pure]
+    [Reads(ReadsAttribute.Reads.Nothing)]
+    public override bool Equals(object obj)
+    {
+      if (obj is FieldAccess fieldAccess)
+      {
+        return Selector.Equals(fieldAccess.Selector);
+      }
+      return false;
+    }
+
+    [Pure]
+    public override int GetHashCode()
+    {
+      return 1;
+    }
+
+    public void Emit(IList<Expr> args, TokenTextWriter stream, int contextBindingStrength, bool fragileContext)
+    {
+      const int opBindingStrength = 0x90;
+      bool parensNeeded = opBindingStrength < contextBindingStrength ||
+                          (fragileContext && opBindingStrength == contextBindingStrength);
+      stream.SetToken(this);
+      Contract.Assert(args.Count == 1);
+      stream.push();
+      if (parensNeeded)
+      {
+        stream.Write("(");
+      }
+      cce.NonNull(args[0]).Emit(stream, 0x00, false);
```

Comment:
The recursive call should pass in `opBindingStrength`, not `0x00`.

Suppose that an operator like `+` (whose `opBindingStrength` is `0x40`) could produce a value of a datatype, then the expression `(a + b)->x` would be printed as `a + b->x` if you pass in `0x00`. There aren't any such operators in Boogie today, but the correct parameter value is still `opBindingStrength` (which in your case is `0x90`).

Reply:
Done.